empty-module.cjs.d.ts

```diff
@@ -0,0 +1,4 @@
+// shim for node modules in React Native
+// see https://github.com/facebook/react-native/issues/5079
+type empty = {};
+export default empty;
```

Comment:
do these not get generated by the tsc, instead, what if we gitignore it?

Reply:
They don't because `index-node.cjs.js`, `index-browser.cjs.js` and `empty-module.cjs.js` live outside of the `./lib` folder and are plain JavaScript files.

Reply:
not sure what you mean

Reply:
The package.json points to these entry files, and TypeScript will look for the equivalent but with the `.d.ts` extension.

Reply:
Ideally I'd get rid of those files sitting at the root of the project and point the package.json to the entry points generated in the `./dist` folder but I didn't want to change everything.

Reply:
I mean that tsc generates these files at publish time, why do they need to be committed on git?

Reply:
>  and are plain JavaScript files.

I see now, makes sense! in the future they should be moved to ts

Reply:
Got it. I will work on it next time.

Reply:
Another thing is, I'd use the `ts-expect` package against the build output to test that the types are emitted properly. That would ensure that the chain `package.json --> enty-*.js --> entry-*.d.ts` is working properly. 

Because right now, there are unit test for the source code, but there's nothing that checks that the build output is in good health.

Reply:
Wasn't aware of that. Thanks for the information. I will check it out :)